### PR TITLE
[Improve] Use new Factory API to create KafkaSource

### PR DESCRIPTION
**Pull request by ruanwenjun**
### Purpose of this pull request

KafkaSource use new Factory API to create, after this change the CatalogTable can be send to post task.

### How was this patch tested?
Use e2e case to assert the catalog generate by KafkaSource

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSource.java [hunk 1]
```diff
@@ -105,127 +67,36 @@ public String getPluginName() {
 
     @Override
     public void prepare(Config config) throws PrepareFailException {
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSource.java]: remove this method ？

https://github.com/apache/seatunnel/pull/5625

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSource.java [hunk 2]
```diff
@@ -105,127 +67,36 @@ public String getPluginName() {
...
-                    break;
-            }
-        }
-
-        setDeserialization(config);
+        kafkaSourceConfig = new KafkaSourceConfig(ReadonlyConfig.fromConfig(config));
     }
 
     @Override
     public SeaTunnelRowType getProducedType() {
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSource.java]: remove this method ？

https://github.com/apache/seatunnel/pull/5625

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/ReadonlyConfig.java [hunk 3]
```diff
@@ -66,6 +73,10 @@ public <T> T get(Option<T> option) {
         return getOptional(option).orElseGet(option::defaultValue);
     }
 
+    public <T> T get(Option<T> option, T defaultValue) {
```
Comment (hailin0) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/ReadonlyConfig.java]: Why not use the following:

1. Define default value in Option

2. getOptional(Option).orElse(defaultValue)

Comment (ruanwenjun) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/ReadonlyConfig.java]: Yes, it's better way, I removed this method.

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceConfig.java [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+    private ConsumerMetadata createConsumerMetadata(ReadonlyConfig readonlyConfig) {
+        CheckResult result = readonlyConfig.checkAllExists(TOPIC, BOOTSTRAP_SERVERS);
+        if (!result.isSuccess()) {
+            throw new KafkaConnectorException(
+                    SeaTunnelAPIErrorCode.CONFIG_VALIDATION_FAILED,
+                    String.format(
+                            "PluginName: %s, PluginType: %s, Message: %s",
+                            CONNECTOR_IDENTITY, PluginType.SOURCE, result.getMsg()));
+        }
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceConfig.java]: I think `checkAllExists` is unnecessary. As `ReadonlyConfig`, if it implement `TableSourceFactory`, we will check whether all configs are legal. Please refer https://github.com/apache/seatunnel/blob/aefc135195f5a10e37aec84e0b7c29708342cfcd/seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/FactoryUtil.java#L116

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/resources/textFormatIT/kafka_source_text_to_console.conf [hunk 5]
```diff
@@ -35,23 +35,84 @@ source {
...
-        c_smallint = smallint
-        c_int = int
-        c_bigint = bigint
-        c_float = float
-        c_double = double
-        c_decimal = "decimal(2, 1)"
-        c_bytes = bytes
-        c_date = date
-        c_timestamp = timestamp
+      columns = [
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/resources/textFormatIT/kafka_source_text_to_console.conf]: Could you create new conf not replace old one? I think we should make sure both two config way worked fine for now.

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSource.java (hunk 2 above)
Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSource.java]: This method can not be removed for now. I need to do some extra work on the translation layer in the next PR before I can properly remove it. For this PR, it is better to keep it.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/resources/textFormatIT/kafka_source_text_to_console.conf (hunk 5 above)
Comment (hailin0) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/resources/textFormatIT/kafka_source_text_to_console.conf]: +1

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSource.java (hunk 1 above)
Comment (hailin0) [reply seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSource.java]: https://github.com/apache/seatunnel/pull/5647/files#diff-693a7a9c81078360435112eee7dca72a70a267a56778d1cfa44aad93033e89abR42

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/resources/textFormatIT/kafka_source_text_to_console.conf (hunk 5 above)
Comment (ruanwenjun) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/resources/textFormatIT/kafka_source_text_to_console.conf]: Agree with you

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceConfig.java [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+                                            0,
+                                            false,
+                                            null,
+                                            null))
+                            .build();
+            return CatalogTable.of(
+                    tableIdentifier,
+                    tableSchema,
+                    Collections.emptyMap(),
+                    Collections.emptyList(),
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceConfig.java:191]: It should be fine after https://github.com/apache/seatunnel/pull/5681 merged.

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]
LGTM

**Pull request merged**